File: C/Newton’s Interpolation.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
main()
{
	int n,i,col,row,j;
	float*x,*y,**table;
	float s,x1,l;
	FILE* fp;
	fp=fopen("data.dat","r");
	printf("input the amounts of datas:");
	fscanf(fp,"%d",&n);
	printf("%d\n",n);
	x=(float*)calloc(n,sizeof(float));
	y=(float*)calloc(n,sizeof(float));
	table=(float**)calloc(n,sizeof(float*));
	for(i=0;i<n;i++)
		table[i]=(float*)calloc(n,sizeof(float));
	printf("input the interpolation datas x,y:\n");
	for(i=0;i<n;i++)
		for(j=0;j<n;j++)
			table[i][j]=0;
	for(i=0;i<n;i++)
		fscanf(fp,"%f,%f",&x[i],&y[i]),printf("x=%f,y=%f\n",x[i],y[i]);
    for(i=0;i<n;i++)
        table[i][0]=y[i];
	printf("input x:");
    fscanf(fp,"%f\n",&x1);
	printf("%f\n",x1);
	for(col=1;col<n;col++)
		for(row=col;row<n;row++)
			table[row][col]=(table[row][col-1]-table[row-1][col-1])/(x[row]-x[row-col]);
	s=table[0][0];
    for(i=1;i<n;i++)
	{   l=1;
		for(j=0;j<i;j++)
			l*=(x1-x[j]);
		l*=table[i][i];
		s+=l;
	}
	fclose(fp); fp=fopen("output.dat","w");
	fprintf(fp,"the interpolation result is: %f",s);
	printf("the interpolation result is: %f\n",s);
	fclose(fp);
}

File: Fortran/Newton’s Interpolation.f90
program NewtonsInterpolation
implicit none
integer::n,i,col,row,j
real,allocatable::x(:),y(:),table(:,:)
real::x1,s,l
open(1,file='data.dat')
open(2,file='output.dat')
print*,"input the amounts of datas:"
read(1,*)n
print*,n
allocate(x(n),y(n),table(n,n))
write(*,'("input the interpolation data x,y")')
do i=1,n
  read(1,*)x(i),y(i)
  write(*,'("x=",f,"y=",f)')x(i),y(i)
end do
print*,"input x:"
read(1,*)x1
print*,x1
table=0
do i=1,n
  table(i,1)=y(i)
end do
do col=2,n
  do row=col,n
    table(row,col)=(table(row,col-1)-table(row-1,col-1))/(x(row)-x(row-col+1))
  end do
end do
s=table(1,1)
do i=2,n
  l=1
  do j=1,i-1
    l=l*(x1-x(j))
  end do
  l=l*table(i,i)
  s=s+l
end do
write(2,'("the interpolation result is:",f)')s
write(*,'("the interpolation result is:",f)')s
close(1)
close(2)
end program